lib/vehicle.rb

```diff
@@ -1,25 +1,53 @@
-require 'date'
-
 class Vehicle
   attr_reader :vin,
               :year,
               :make,
               :model,
-              :engine
+              :engine,
+              :registration_date,
+              :plate_type
 
   def initialize(vehicle_details)
     @vin = vehicle_details[:vin]
     @year = vehicle_details[:year]
     @make = vehicle_details[:make]
     @model = vehicle_details[:model]
     @engine = vehicle_details[:engine]
+    @registration_date = vehicle_details[:registration_date]
+    @plate_type = vehicle_details[:plate_type]
   end
 
+  
   def antique?
     Date.today.year - @year > 25
   end
 
   def electric_vehicle?
     @engine == :ev
   end
+
+  def registration_date(date = nil)
+      return nil if date.nil?
```

Comment:
this is called a guard clause and is helpful to check for conditions before running any additional code. Nice job adding one.

Reply:
For this method, just doing Date.today would have worked. :) Glad you got to play with .strftime though!